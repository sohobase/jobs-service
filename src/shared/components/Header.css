@import "../styles/fonts.css";
@import "../styles/variables.css";

.header {
  &, & .wrapper {
    height: var(--header-height);
  }

  & .wrapper {
    display: flex;
    align-items: center;
    /* @TODO: this .wrapper properties have to be inherit from a common */
    max-width: var(--content-max-width);
    margin: 0 auto;
  }

  & .brand {
    font-size: var(--font-size-extra-large);
    font-weight: var(--font-weight-extra-bold);
  }

  & .navigation {
    flex-grow: 1;
    display: flex;
    align-items: flex-end;
    align-content: flex-end;
    justify-content: flex-end;
    font-size: var(--font-size-small);

    & > * {
      margin-left: var(--offset);
    }
  }
}
