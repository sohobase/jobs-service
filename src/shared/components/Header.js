import React from 'react';
import { Link } from 'react-router';
import style from './Header.css';

export default () => (
  <header className={style.header}>
    <div className={style.wrapper}>
      <Link to="/">
        <img src="/static/img/logo.png" className={style.brand} />
      </Link>
      <nav className={style.navigation}>
        <Link to="/help">Ask us anything</Link>
        <Link to="/job/1234">Example Job</Link>
        <Link to="/post">Post new offer</Link>
      </nav>
    </div>
  </header>
);
