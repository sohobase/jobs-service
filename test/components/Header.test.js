import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../src/shared/components/Header';

describe('<Header />', () => {
  // -- Renders
  it('renders by default', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
