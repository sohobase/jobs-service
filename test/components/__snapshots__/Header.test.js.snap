exports[`<Header /> renders by default 1`] = `
<header
  className="header">
  <div
    className="wrapper">
    <a
      onClick={[Function]}
      style={Object {}}>
      <img
        alt=""
        className="brand"
        src="/static/img/logo.png" />
    </a>
    <nav
      className="navigation">
      <a
        onClick={[Function]}
        style={Object {}}>
        Ask us anything
      </a>
      <a
        onClick={[Function]}
        style={Object {}}>
        Example Job
      </a>
      <button
        className="button accent"
        onClick={undefined}>
        Post new offer
      </button>
    </nav>
  </div>
</header>
`;
